docs/Beginner-2/README.md

```diff
@@ -0,0 +1,481 @@
+# C Programming Foundations
+
+## Introduction
+
+Welcome to the world of C programming! In this guide, we'll cover essential concepts and problem-solving techniques that will set the foundation for your journey into the programming realm.
+
+## Table of Contents
+1. [Variables](#variables-in-c)
+   - [What are Variables?](#what-are-variables)
+2. [Loops](#loops-in-c)
+   - [What are Loops?](#what-are-loops)
+   - [For Loop](#for-loop-explanation)
+   - [Problem: Factorial Calculation](#problem-factorial-calculation)
+   - [While Loop](#While-loop-explanation)
+   - [Problem: Print N numbers](#problem-print-N-numbers)
+   - [Do-While Loop](#Do-While-explanation)
+   - [Problem:  Add numbers entered by user](#Problem-Add-numbers-entered-by-user)
+   - [Switch...Case](#Switch-case-explanation)
+   - [Problem: Simple Calculator](#example)
+3. [Conditional Statements](#conditional-statements-in-c)
+   - [What are Conditional Statements?](#what-are-conditional-statements)
+   - [if-else statement](#explanation)
+   - [Problem: Leap Year Checker](#problem-leap-year-checker)
+   - [if-else ladder](#explanation-problem)
+   - [Nested if-else](#explanation-problem)
+4. [Arrays](#arrays-in-c)
+   - [What are Arrays?](#what-are-arrays)
+   - [Problem: Largest Element Finder](#problem-find-the-largest-element)
+  5. [Past Year Questions](#Past-Year-Questions)
+
+## Variables in C
+
+### What are Variables?
+
+In programming, variables are containers that store data. They have a type and a name.
+
+```ruby
+int age;  // Declaration
+age = 21; // Assignment
+```
+**Note:** The value of a variable can be changed, hence the name variable.
+
+```ruby
+char ch = 'a';
+ch = 'l';
+```
+
+C is a strongly typed language. This means that the variable type cannot be changed once it is declared. For example:
+
+```ruby
+int number = 5;     // integer variable
+number = 5.5;       // error, can't assign float to int
+double number;      // error, can't redefine the data type 
+```
+
+## Loops in C
+
+### What are Loops?
+
+In programming, a loop is used to repeat a block of code until the specified condition is met.
+
+### For Loop
+```ruby
+for (int i = 0; i < 5; i++) {
+    // Code inside the loop
+}
+```
+![For Loop Working](https://cdn.programiz.com/sites/tutorial2program/files/c-for-loop.jpg)
+<br>
+*Working of For Loop*
+
+### Problem: Factorial Calculation
+
+Write a program to find the factorial of a number.
+```c
+#include <stdio.h>
+
+int main() {
+    int number, factorial = 1;
+    printf("Enter a number: ");
```

Comment:
SUGGESTION: Although people know about `printf` and `scanf`, and you will explain it during the session, you can add some lines in the docs or even as a comment just mentioning what it is. To make the docs more robust.

Reply:
Ok, will add the necessary comments